### ISSUE TITLE
Retriever Notifications

### ISSUE THREAD

**Issue by stdavis** [agrc/roadkill-mobile#35]
Set up some sort of notification system ([maybe pub/sub or cloud task?](https://docs.google.com/document/d/1ov5tFX-dcNkqvn7aCzUCsaMyhyBOdDq5ECVPCN-0o54/edit#heading=h.ti10hi3lro96)).

Notifications need to be sent to retrievers based on the retriever service areas polygons. They should be emails (perhaps with a text option as an alternative).

**Comment by stdavis** [comment]
@nathan: Some notification questions:
* Can I assume that all users who need to receive notifications will be registered with UtahID and this application?
* Will there be the possibility of multiple users assigned to a single area?
* Will there be the possibility of multiple areas assigned to a single user?

**Comment by nathankota** [comment]
> Can I assume that all users who need to receive notifications will be registered with UtahID and this application?

Yes, Agency and Contractor users will both be registered with UtahID.

> Will there be the possibility of multiple users assigned to a single area?

Yes

> Will there be the possibility of multiple areas assigned to a single user?

Yes

**Comment by stdavis** [comment]
@nathankota Here's another question for the client: What do they want to happen if a report is submitted outside of any notification area?

Should I:
1. Send the notification email to the admin email group (`wvc-dnr-dwr-notify@utah.gov`)
2. Not send any notification email
3. Something else?

**Comment by nathankota** [comment]
> @nathankota Here's another question for the client: What do they want to happen if a report is submitted outside of any notification area?

I have reached out to the client about this. It may not be an issue as long as the DWR and UDOT boundaries provide full coverage of the state, and always have a contact associated with them. Of course, contractor contacts will only be notified where there is a contactor boundary with an associated contact.

**Comment by nathankota** [comment]
1. Send the notification email to the admin email group (`wvc-dnr-dwr-notify@utah.gov`)
> @nathankota Here's another question for the client: What do they want to happen if a report is submitted outside of any notification area?
> 
> Should I:
> 
> 1. Send the notification email to the admin email group (`wvc-dnr-dwr-notify@utah.gov`)
> 2. Not send any notification email
> 3. Something else?

**Comment by stdavis** [comment]
This is implemented in staging. The only missing piece is to populate the `notification_areas` polygon layer and `users_have_notification_areas` relationship table. I'm not sure who they would like to do that. Maybe Jessie? There's some test data in them currently that can be deleted when they have some real data to put in there.

**Comment by nathankota** [comment]
Checked in with clients again on the status of the notification data.
